packages/itwin/property-grid/src/components/Header.tsx

```diff
@@ -6,32 +6,105 @@
 export interface HeaderProps {
   className?: string;
   onBackButtonClick?: () => void;
+  title?: React.ReactNode;
+  headerTools?: React.ReactNode;
+  onSearchStringChange?: (searchInput: string) => void;
 }
 
-/** @internal */
-export function Header({ className, children, onBackButtonClick }: PropsWithChildren<HeaderProps>) {
-  return <div className={classnames("property-grid-react-panel-header", className)}>
-    {
-      onBackButtonClick
-        ? <IconButton
-          styleType="borderless"
-          onClick={onBackButtonClick}
-          title={PropertyGridManager.translate("header.back")}
-          className="property-grid-react-header-back-button"
-        >
-          <SvgProgressBackwardCircular />
-        </IconButton>
-        : null
+export interface DebouncedSearchBoxProps {
+  onSearchStringChange: (searchInput: string) => void;
+  onClose: () => void;
+  onOpen: () => void;
+  className: string;
+}
+
+export function DebouncedSearchBox({ onSearchStringChange, onClose, onOpen, className }: DebouncedSearchBoxProps){
+  const [inputValue, setInputValue] = useState<string>("");
+  const onSearchStringChangeRef = useRef(onSearchStringChange);
+  onSearchStringChangeRef.current = onSearchStringChange;
+
+  useEffect(() => {
+    if (!inputValue) {
+      onSearchStringChangeRef.current("");
+      return;
     }
-    {children}
+
+    const timeoutId = setTimeout(() => {
+      onSearchStringChangeRef.current(inputValue);
+    }, 25);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [inputValue]);
+
+  return <SearchBox
+    expandable
+    onCollapse={onClose}
+    onExpand={onOpen}
+    className={className}
+  >
+    <SearchBox.CollapsedState>
+      <SearchBox.ExpandButton
+        title="Open search bar"
+      />
+    </SearchBox.CollapsedState>
+    <SearchBox.ExpandedState >
+      <SearchBox.Input
+        placeholder={"Search"}
+        onChange={(e) => setInputValue(e.currentTarget.value)}
+      />
+      <SearchBox.CollapseButton
+        title="Close search bar"
+        onClick={() => {
+          setInputValue("");
+        }}
+      />
+    </SearchBox.ExpandedState>
+  </SearchBox>;
+}
+
+/** @internal */
+export function Header({ className, onBackButtonClick, headerTools, title, onSearchStringChange }: PropsWithChildren<HeaderProps>) {
+  const [searchBarIsExpanded, setSearchBarIsExpanded] = useState(false);
+
+  return <div className={classnames("property-grid-react-panel-header", searchBarIsExpanded && "search-bar-expanded", className)}>
+    <div className="header-title">
+      {
+        onBackButtonClick
+          ? <IconButton
+            styleType="borderless"
+            onClick={onBackButtonClick}
+            title={PropertyGridManager.translate("header.back")}
+            className="property-grid-react-header-back-button"
+          >
+            <SvgProgressBackwardCircular />
+          </IconButton>
+          : null
+      }
+      <div className={"header-text"}>
```

Comment:
Looks like this `div` is necessary only when using `Header` in `PropertyGridHeader`. I thinks `title` passed from `PropertyGridHeader` can be wrapped into `div` instead of fragment (`<>`).